src/services/blockchain/clients/midnight/client.rs

```diff
@@ -0,0 +1,246 @@
+
+/// Extended functionality specific to Midnight blockchain
+#[async_trait]
+pub trait MidnightClientTrait {
+	/// Retrieves a transaction receipt by its hash
+	///
+	/// # Arguments
+	/// * `transaction_hash` - The hash of the transaction to look up
+	///
+	/// # Returns
+	/// * `Result<TransactionReceipt, anyhow::Error>` - Transaction receipt or error
+	async fn get_transaction_receipt(
+		&self,
+		transaction_hash: String,
+	) -> Result<EVMTransactionReceipt, anyhow::Error>;
+
+	/// Retrieves logs for a range of blocks
+	///
+	/// # Arguments
+	/// * `from_block` - Starting block number
+	/// * `to_block` - Ending block number
+	///
+	/// # Returns
+	/// * `Result<Vec<Log>, anyhow::Error>` - Collection of matching logs or error
+	async fn get_logs_for_blocks(
+		&self,
+		from_block: u64,
+		to_block: u64,
+	) -> Result<Vec<EVMReceiptLog>, anyhow::Error>;
+}
+
+#[async_trait]
+impl<T: Send + Sync + Clone + BlockchainTransport> MidnightClientTrait for MidnightClient<T> {
+	/// Retrieves a transaction receipt by hash with proper error handling
+	#[instrument(skip(self), fields(transaction_hash))]
+	async fn get_transaction_receipt(
+		&self,
+		transaction_hash: String,
+	) -> Result<EVMTransactionReceipt, anyhow::Error> {
+		let hash = string_to_h256(&transaction_hash)
+			.map_err(|e| anyhow::anyhow!("Invalid transaction hash: {}", e))?;
+
+		let params = json!([format!("0x{:x}", hash)])
+			.as_array()
+			.with_context(|| "Failed to create JSON-RPC params array")?
+			.to_vec();
+
+		let response = self
+			.http_client
+			.send_raw_request(
+				"eth_getTransactionReceipt",
+				Some(serde_json::Value::Array(params)),
+			)
+			.await
+			.with_context(|| format!("Failed to get transaction receipt: {}", transaction_hash))?;
+
+		// Extract the "result" field from the JSON-RPC response
+		let receipt_data = response
+			.get("result")
+			.with_context(|| "Missing 'result' field")?;
+
+		// Handle null response case
+		if receipt_data.is_null() {
+			return Err(anyhow::anyhow!("Transaction receipt not found"));
+		}
+
+		Ok(serde_json::from_value(receipt_data.clone())
+			.with_context(|| "Failed to parse transaction receipt")?)
+	}
+
+	/// Retrieves logs within the specified block range
+	///
+	/// # Arguments
+	/// * `from_block` - Starting block number
+	/// * `to_block` - Ending block number
+	///
+	/// # Returns
+	/// * `Result<Vec<EVMReceiptLog>, anyhow::Error>` - Collection of matching logs or error
+	#[instrument(skip(self), fields(from_block, to_block))]
+	async fn get_logs_for_blocks(
+		&self,
+		from_block: u64,
+		to_block: u64,
+	) -> Result<Vec<EVMReceiptLog>, anyhow::Error> {
+		// Convert parameters to JSON-RPC format
+		let params = json!([{
+			"fromBlock": format!("0x{:x}", from_block),
+			"toBlock": format!("0x{:x}", to_block)
+		}])
+		.as_array()
+		.with_context(|| "Failed to create JSON-RPC params array")?
+		.to_vec();
+
+		let response = self
+			.http_client
+			.send_raw_request("eth_getLogs", Some(params))
```

Comment:
Same here, is it valid?

Reply:
Same as: https://github.com/OpenZeppelin/openzeppelin-monitor/pull/175#discussion_r2100140286

Sorry, should've mentioned that in the PR description